asdf/tests/test_schema.py

```diff
@@ -105,74 +111,122 @@ def validate_schema(path):
             if os.path.splitext(fname)[0] in (
                     'draft-01', 'asdf-schema-1.0.0'):
                 continue
-            yield validate_schema, os.path.join(root, fname)
+            yield os.path.join(root, fname)
 
+def test_schema_example(filename, example):
+    """Pytest to check validity of a specific example within schema file
 
-def test_all_schema_examples():
+    Parameters:
+    -----------
+    filename : name of the schema file containing example to be tested
+
+    example: string representing example
+
+    This function is called with a range of parameters by pytest's
+    'parametrize' utility in order to account for all examples in all schema
+    files.
+    """
     # Make sure that the examples in the schema files (and thus the
     # ASDF standard document) are valid.
+    buff = helpers.yaml_to_asdf('example: ' + example.strip())
+    ff = asdf.AsdfFile(uri=util.filepath_to_url(os.path.abspath(filename)))
+
+    # Fake an external file
+    ff2 = asdf.AsdfFile({'data': np.empty((1024*1024*8), dtype=np.uint8)})
+
+    ff._external_asdf_by_uri[
+        util.filepath_to_url(
+            os.path.abspath(
+                os.path.join(
+                    os.path.dirname(filename), 'external.asdf')))] = ff2
+
+    # Add some dummy blocks so that the ndarray examples work
+    for i in range(3):
+        b = block.Block(np.empty((1024*1024*8), dtype=np.uint8))
+        b._used = True
+        ff.blocks.add(b)
+    b._array_storage = "streamed"
+
+    try:
+        # Ignore warnings that result from examples from schemas that have
+        # versions higher than the current standard version.
+        # TODO: this should no longer be necessary once the library can
+        # actually account for higher versions
```

Comment:
When updating the schemas to 1.1.0 I left the check for the version as is but I suspect this is not what was intended originally. It would be good to actually know what the intention was.